### PR TITLE
NLP - Stemming Directive

### PR DESCRIPTION
**Pull request by nitinmotgi**

### CODE REVIEW

File: docs/directives/stemming.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Write as JSON Map
```
Comment (sreevatsanraman) [inline docs/directives/stemming.md]: This should be NLP stemming?

File: wrangler-core/src/main/java/co/cask/wrangler/steps/nlp/Stemming.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+package co.cask.wrangler.steps.nlp;
+
+import co.cask.wrangler.api.AbstractStep;
+import co.cask.wrangler.api.PipelineContext;
+import co.cask.wrangler.api.Record;
+import co.cask.wrangler.api.StepException;
+import co.cask.wrangler.api.Usage;
+import co.cask.wrangler.steps.nlp.internal.PorterStemmer;
+import edu.emory.mathcs.backport.java.util.Arrays;
```
Comment (sreevatsanraman) [inline wrangler-core/src/main/java/co/cask/wrangler/steps/nlp/Stemming.java]: Should this be java.util.Arrays instead?

File: wrangler-core/src/main/java/co/cask/wrangler/steps/nlp/internal/PorterStemmer.java [hunk 3]
```diff
@@ -0,0 +1,423 @@
+/*
```
Comment (sreevatsanraman) [inline wrangler-core/src/main/java/co/cask/wrangler/steps/nlp/internal/PorterStemmer.java:1]: Add license from the code that is copied from Stanford NLP.

File: wrangler-core/src/test/java/co/cask/wrangler/steps/nlp/StemmingTest.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    );
+
+    records = PipelineTest.execute(directives, records);
+
+    Assert.assertTrue(records.size() == 1);
+    Assert.assertEquals(Arrays.asList("how", "ar", "you", "do", "do", "you", "have", "appl"),
+                        records.get(0).getValue("words_porter"));
+  }
+
+}
```
Comment (sreevatsanraman) [inline wrangler-core/src/test/java/co/cask/wrangler/steps/nlp/StemmingTest.java]: new line.

**Review by sreevatsanraman — CHANGES_REQUESTED** [review]
Minor comments, fix and merge.

File: wrangler-core/src/main/java/co/cask/wrangler/steps/nlp/Stemming.java (hunk 2 above)
Comment (nitinmotgi) [reply wrangler-core/src/main/java/co/cask/wrangler/steps/nlp/Stemming.java]: fixed

File: wrangler-core/src/main/java/co/cask/wrangler/steps/nlp/internal/PorterStemmer.java (hunk 3 above)
Comment (nitinmotgi) [reply wrangler-core/src/main/java/co/cask/wrangler/steps/nlp/internal/PorterStemmer.java:1]: fixed

File: wrangler-core/src/test/java/co/cask/wrangler/steps/nlp/StemmingTest.java (hunk 4 above)
Comment (nitinmotgi) [reply wrangler-core/src/test/java/co/cask/wrangler/steps/nlp/StemmingTest.java]: done

**Pull request merged**